tabular/src/autogluon/tabular/utils.py

```diff
@@ -150,3 +156,243 @@ def infer_eval_metric(problem_type: str) -> Scorer:
+        This is necessary to compute importance of original data features in `X` prior to their transformation assuming `predict_func` does not perform the transformation already.
+            Example: `transform_func` is necessary to compute the importance of a text feature prior to being transformed into ngrams by `transform_func` when `predict_func` expects ngram features as input.
+    transform_func_kwargs : dict, default {}
+        Keyword arguments to be appended to calls to `transform_func(X, **kwargs)`.
+    time_limit : float, default None
+        Time in seconds to limit the calculation of feature importance.
+        If None, feature importance will calculate without early stopping.
+        A minimum of 1 full shuffle set will always be evaluated. If a shuffle set evaluation takes longer than `time_limit`, the method will take the length of a shuffle set evaluation to return regardless of the `time_limit`.
+        If `num_shuffle_sets==1`, `time_limit` will be ignored.
+    silent : bool, default False
+        Whether to suppress logging output.
+    log_prefix : str, default ''
+        Prefix to add to logging statements.
+
+    Returns
+    -------
+    Pandas `pandas.DataFrame` of feature importance scores with 3 columns:
+        index: The feature name.
+        'importance': The feature importance score.
+        'stddev': The standard deviation of the feature importance score. If NaN, then not enough num_shuffle_sets were used to calculate a variance.
+        'z_score': The z-score of the feature importance score. Equivalent to 'importance' / 'stddev'.
+            A z-score of +4 or higher indicates that the feature is almost certainly useful and should be kept.
+            A z-score of +2 indicates that the feature has a 97.5% chance of improving model quality when present.
+            A z-score that is 0 or negative indicates that the feature can likely be dropped without negative impact to model quality.
+            A z-score of NaN indicates that the feature's stddev was NaN or that the model predictions were never impacted by the feature (importance 0 and stddev 0). This indicates that the feature can safely be dropped.
+            A z-score of +inf or -inf indicates that `subsample_size` and/or `num_shuffle_sets` were too small to calculate variance for the feature (non-zero importance with zero stddev).
+    """
+    if num_shuffle_sets is None:
+        num_shuffle_sets = 1 if time_limit is None else 10
+
+    time_start = time.time()
+    if predict_func_kwargs is None:
+        predict_func_kwargs = dict()
+    if transform_func_kwargs is None:
+        transform_func_kwargs = dict()
+    if features is None:
+        features = list(X.columns)
+    num_features = len(features)
+
+    if subsample_size and (len(X) > subsample_size):
+        # TODO: Stratify? We currently don't know in this function the problem_type (could pass as additional arg).
+        X = X.sample(subsample_size, random_state=0)
```

Comment:
Shouldn't the subsample be re-drawn for each `shuffle_repeat in  `1:num_shuffle_sets`?

Otherwise the std-dev doesn't necessarily go up (as desired) when `subsample_size` is small, right?

Reply:
I don't believe that will change std-dev scale, but it is a good idea to increase the diversity of the subsamples, added.

Reply:
> > It's a little weird that by default, `stddev` and `z_score` will be NaN (I expect most users to just use default and then be confused by the NaN). Maybe default value of `num_shuffle_sets` can be slightly higher and `subsample_size` accordingly reduced to compensate?
> 
> My main concern is for the expensive runs. Reducing subsample_size < 1000 doesn't necessarily speed up the run a great deal since we are batching the feature subsamples, and `num_shuffle_sets==2` gives very large error bands. Anything >2 will be time consuming in some cases, so we should probably see if anyone has issues with runtime after this change. If nobody complains, then we can maybe set num_shuffle_sets==3 instead of 1 by default.

I think `num_shuffle_sets == 3` by default sounds reasonable. Or we could set time-limits by default. Or we could just remove all of the NaN columns in the returned output when we see that `num_shuffle_sets == 1` (to avoid user confusion), and stick with this default.

Reply:
added num_shuffle_sets==3 as default